lib/ReaderCollection.js

```diff
@@ -47,21 +51,20 @@ class ReaderCollection extends AbstractReader {
 	 * @param {string} virPath Virtual path
 	 * @param {object} options Options
 	 * @param {@ui5/fs/tracing.Trace} trace Trace instance
-	 * @returns {Promise<@ui5/fs/Resource>} Promise resolving to a single resource
+	 * @returns {Promise<@ui5/fs/Resource|null>}
+	 *   Promise resolving to a single resource or <code>null</code> if no resource is found
 	 */
 	_byPath(virPath, options, trace) {
 		const that = this;
 		const resourceLocatorCount = this._readers.length;
 		let resolveCount = 0;
 
-		if (this._readers.length === 0) {
-			// Promise.race doesn't resolve for empty arrays
-			return Promise.resolve();
+		if (resourceLocatorCount === 0) {
+			// Short-circuit if there are no readers (Promise.race does not settle for empty arrays)
+			return Promise.resolve(null);
```

Comment:
`trace.collection(that._name);` should be called here, right?

Reply:
Ups, done